plugin/src/main/java/at/helpch/chatchat/util/MessageProcessor.java

```diff
@@ -59,10 +59,23 @@ private MessageProcessor() {
         throw new AssertionError("Util classes are not to be instantiated!");
     }
 
+    /**
+     * Process a message for a user and send it to the recipients.
+     * @param plugin The plugin instance.
+     * @param user The user sending the message.
+     * @param channel The channel the user is sending the message to.
+     * @param previousChannel The previous channel the user was in. Used to switch back to after sending the message.
+     *                        Used to switch back when sending a message to a channel with a prefix.
+     * @param message The message to send.
+     * @param async Whether to process the message asynchronously.
+     *
+     * @return Whether the message was sent successfully.
+     */
     public static boolean process(
         @NotNull final ChatChatPlugin plugin,
         @NotNull final ChatUser user,
         @NotNull final Channel channel,
+        @NotNull final Channel previousChannel,
```

Comment:
I believe it is best to let each processor caller deal with what the previous user channel should be while the function would only let them know that the user channel will be temporarily changed during this function call but reverted at the end.

This function is already very complex so we should not add more complexity to it.

In practice, this means rolling back the changes to this function and always setting back the old channel of the user in ChatListener.java